eureka2-read-server/src/main/java/com/netflix/eureka2/server/registry/EurekaReadServerRegistry.java

```diff
@@ -83,19 +87,9 @@ public int size() {
         throw new UnsupportedOperationException("method not supported by EurekaReadServerRegistry");
     }
 
-    // TODO As read server is based on client API, and is doing fullRegistryFetch from write server, we cannot generate batch markers4
-    // TODO This is one of the reasons why we cannot depend on Eureka client API in the read server.
     @Override
     public Observable<ChangeNotification<InstanceInfo>> forInterest(final Interest<InstanceInfo> interest) {
-        return eurekaClient.forInterest(interest).map(new Func1<ChangeNotification<InstanceInfo>, ChangeNotification<InstanceInfo>>() {
-            @Override
-            public ChangeNotification<InstanceInfo> call(ChangeNotification<InstanceInfo> notification) {
-                if (notification.isDataNotification()) {
-                    return notification;
-                }
-                return null;
-            }
-        }).filter(filterNullValuesFunc());
+        return interestClient.forInterest(interest).flatMap(bufferStartEndDelineateFun(interest));
```

Comment:
I have added a comment. It would be actually good to have some flexibility in choosing which markers we prefer on registry/client API as technically we have both implementations. Possibly improvement in the future.